JobScheduler.js

```diff
@@ -147,26 +148,37 @@ class JobScheduler {
             }
             return resolve();
           } else {
-            logger.info('+-> System is in maintenance or last maintenance operation has failed :', _.pick(maintenanceInfo, ['progress', 'state', 'completedAt', 'reason', 'toVersion', 'fromVersion', 'createdAt', 'updatedAt', 'createdBy', 'updatedBy']));
-            const currTime = moment();
-            const maintenanceStartTime = _.get(maintenanceInfo, 'createdAt');
-            if (maintenanceStartTime && currTime.diff(maintenanceStartTime) > config.scheduler.maintenance_mode_time_out) {
-              logger.warn(`System in maintenance since ${maintenanceInfo.createdAt}. Exceeds configured maintenance timeout. Flagging the current maintenance window as aborted.`);
-              const success = resolve;
-              return maintenanceManager
-                .updateMaintenace(`System in maintenance beyond configured timeout time ${config.scheduler.maintenance_mode_time_out/1000/60} (mins)`,
-                  CONST.OPERATION.ABORTED,
-                  CONST.SYSTEM_USER)
-                .then(success)
-                .catch((err) => {
-                  logger.error('error occurred while updating maintenance info', err);
-                  reject(err);
-                })
-                .finally(() => {
-                  if (this.intervalTimer) {
+            const maintInfoAttrs = ['progress', 'state', 'completedAt', 'reason', 'toVersion', 'fromVersion', 'createdAt', 'updatedAt', 'createdBy', 'updatedBy'];
+            if (_.get(maintenanceInfo, 'state', '') !== CONST.OPERATION.IN_PROGRESS) {
+              logger.info(`+-> System is not in maintenance, but its current state is: ${_.get(maintenanceInfo, 'state', '')}, not as expected. `);
```

Comment:
just for my understanding , at this point we cannot say that this state is not as expected right :) ?

Only if SF is up, we can say that , correct ? - which we check right below.